neural_compressor/torch/quantization/algorithm_entry.py

```diff
@@ -67,6 +73,52 @@ def gptq_entry(
     return model
 
 
+###################### HHQ Algo Entry ##################################
+
+
+def _convert_hqq_module_config(config: HQQConfig) -> HQQModuleConfig:
+    nbits = config.nbits
+    group_size = config.group_size
+    quant_zero = config.quant_zero
+    quant_scale = config.quant_scale
+    scale_quant_group_size = config.scale_quant_group_size
+
+    weight_qconfig = QTensorConfig(
+        nbits=nbits, channel_wise=True, group_size=group_size, optimize=True, round_zero=True if nbits == 4 else False
+    )
+    zero_qconfig = None
+    if quant_zero:
+        zero_qconfig = QTensorConfig(nbits=8, channel_wise=False, group_size=None, optimize=False)
+    scale_qconfig = None
+    if quant_scale:
+        scale_qconfig = QTensorConfig(nbits=8, channel_wise=True, group_size=scale_quant_group_size, optimize=False)
+    hqq_module_config = HQQModuleConfig(weight=weight_qconfig, scale=scale_qconfig, zero=zero_qconfig)
+    logger.debug(hqq_module_config)
+    return hqq_module_config
+
+
+def _parse_hqq_configs_mapping(configs_mapping: Dict[OP_NAME_AND_TYPE_TUPLE_TYPE, HQQConfig]):
+    qconfig_mapping = {}
+    for (op_name, op_type), quant_config in configs_mapping.items():
+        if quant_config.skip_lm_head and "lm_head" in op_name:
+            logger.warning("Skip quantizing %s due to `skip_lm_head` is True.", op_name)
+            continue
+        qconfig_mapping[op_name] = _convert_hqq_module_config(quant_config)
+    return qconfig_mapping
+
+
+@register_algo(name=HQQ)
+@torch.no_grad()
+def hqq_entry(
+    model: torch.nn.Module, configs_mapping: Dict[Tuple[str, Callable], HQQConfig], *args, **kwargs
+) -> torch.nn.Module:
+    logger.info("Quantize model with the HQQ algorithm.")
+    qconfig_mapping = _parse_hqq_configs_mapping(configs_mapping)
```

Comment:
How about putting this func to `utility.py` in algo folder? we can keep here clear to read.

Reply:
Thanks, relocated the HQQ-related config transformation to hqq [0623ab3](https://github.com/intel/neural-compressor/pull/1597/commits/0623ab34068aa88e63b3f153c477becd025f174b).